### ISSUE TITLE
General Culling

### ISSUE THREAD

**Issue by kbladin** [OpenSpace/OpenSpace#387]
The `Renderable` base class should be cullable using its bounding sphere. The renderer should test each renderable if it is cullable before rendering it, so when creating a renderable sub class one should not need to worry about culling since it should be done outside of the render function.

Both frustum culling and distance culling should be used. A possibility to look in to would also be occlusion culling using minimum bounding spheres but then things will get a bit more complicated.

A general culling class could be implemented similar to the `ChunkCuller` in globe browsing, however taking other arguments in the `isCullable` function. It could take a bounding sphere as argument and also be extended for other shapes.

**Comment by alexanderbock** [comment]
@jccosta  started implementing that locally for the atmosphere.. Maybe that can be generalized?  I agree that they should be culled.

It's important to keep in mind that we can run multiple cameras for a scene and the culling can be different for each camera

**Comment by kbladin** [comment]
Thats right. As long as the culling uses the correct view projection matrix it should work out the same way for all rendering nodes. Same as is done for chunk culling in globe browsing and I assume for atmosphere culling as well. It would be nice if the person writing a new renderable class does not have to worry about the culling at all.

**Comment by alexanderbock** [comment]
Agreed

**Comment by alexanderbock** [comment]
Potential problems when dealing with multiple cameras, where an object is visible in one camera, but not in another?
